### ISSUE TITLE
Questions about FreeSurfer White Matter Lobe Parcellation and DKT Segmentation Modification

### ISSUE THREAD

**Issue by dat-rohit** [Deep-MI/FastSurfer#634]
## Question/Support Request

Hi everyone,
I am working on obtaining white matter labels parcellated in different brain lobes (frontal, temporal, parietal, etc.). Currently, after running the FastSurfer full pipeline, I use these commands to obtain the end results:

```
# Generate lobe parcellation
mri_annotation2label --subject [SUBJECT] --hemi lh --lobesStrict lobes
mri_annotation2label --subject [SUBJECT] --hemi rh --lobesStrict lobes
mri_aparc2aseg --s [SUBJECT] --labelwm --hypo-as-wm --rip-unknown --volmask --o wmparc.lobes.mgz --ctxseg aparc+aseg.mgz --annot lobes --base-offset 200
```
I have several questions about this process:

**DKT Segmentation Editing:**

- **Is there a way to edit the output aparc.DKTatlas+aseg.deep.mgz segmentation file (produced by the CNN) with small manual corrections before FastSurfer continues with the rest of the pipeline?** I've tried simply replacing the file by my edited version before letting recon-surf.sh run but I am not sure this is the right way to do it
- **Is this DKT file refined later during the pipeline? If yes, is there a way to force the pipeline to keep the segmentation as is?**

**White Matter Parcellation Alignment:**

- When checking wmparc.lobes.mgz, I notice the white matter voxels align perfectly with those segmented in aparc+aseg.mgz (which makes sense as it's used as input to mri_aparc2aseg)
However, when using aparc.DKTatlas+aseg.deep.mgz as input to --ctxseg instead, I observe (check screenshots):
  - The ROIs don't completely align with the newly produced wmparc.lobes.mgz
  - Some GM ROIs get refined
  - Some white matter voxels are left unassigned (shown in green)

- **Is it possible to make the white matter parcellation generation step strictly use the DKT segmentation file as a base?** This would ensure that the lobe-parcellated white matter voxels align and remain consistent with the segmented white matter areas in the input DKT atlas.

Thank you for any insights!
## Screenshots / Log files
1st row: aparc+aseg.mgz segmentation and corresponding wmparc.lobes.mgz

![Image](https://github.com/user-attachments/assets/f2f01fba-d166-4967-bc9f-c9afd2dc5d50)
2nd row: aparc.DKTatlas+aseg.deep.mgz segmentation and corresponding wmparc.lobes.mgz
![Image](https://github.com/user-attachments/assets/c69b4ca3-ad68-4aae-81c1-678e61e5a6d3)

## Environment

- Installation type: native 
- FreeSurfer Version: 7.4.1
- OS: Ubuntu 22.04.5 LTS
- GPU: RTX 2080 Ti / CUDA 12.2
- Python 3.10.10

**Comment by m-reuter** [comment]
Hi, thanks for your interest in FastSurfer. I think we should be able to get you what you want. First about your questions:

1. Editing of deep-learning segmentation. Yes you can edit the segmentation, as we derive brain mask, aseg, etc from it during the segmentation module, it would be good to re-generate those files. In our upcoming release, this will easily be possible. See https://deep-mi.org/FastSurfer/dev/overview/EDITING.html for a description of edits, including asegDKT_segfile edits. If your edits don't really affect subcortical structures or the outer cortex much, you may be able to skip that step (for now) and directly proceed with the surface module. 
2. The deep-learning based segmentation file `aparc.DKTatlas+aseg.deep.mgz` is not refined later, but several files are derived from it. On top of the aseg and brain mask (see above) during segmentation module, also the surface module makes heavy use of that file. The surface stream then creates surfaces and fine-tunes their placement. After that it creates the final segmentation which is corrected by the surfaces `aparc.DKTatlas+aseg.mapped.mgz` and therefore differs slightly from the deep input segmentation. It should be more correct and is the recommended output file. See https://deep-mi.org/FastSurfer/dev/overview/OUTPUT_FILES.html for our documentation on output files. 
3. The reason you see differences is that your mapping uses the surfaces to place the cortical labels inside the WM regions. Only voxels inside the WM surface will be coloured. The original deep learning base segmentation however has a few voxels labeled as WM that are actually outside the WM surface. So just use the `aparc.DKTatlas+aseg.mapped.mgz` as the basis for painting the WM labels and it should work. 
4. It is not really (easily) possible to make the WM parcellation generation strictly use the input deep-learning based segmentation file. The reason is 2. above, that the surfaces are used to paint it in, and they have already moved slightly to better places. One way to achieve this could be to not move surfaces at all, but then you would loose the benefit of fine-tuned grey-white boundaries. Better is to do as described in 3, and use the final and updated files. 

I hope this helps.

**Comment by dat-rohit** [comment]
Hi @m-reuter,
Thank you for your response. I have some follow-up points:

- **Regarding your suggestion to use aparc.DKTatlas+aseg.mapped.mgz:** In my tests, this file appears identical to aparc+aseg.mgz (both show the same refinements compared to aparc.DKTatlas+aseg.deep.mgz). Using either file for mri_aparc2aseg produces identical results. Could you clarify what specific issue this suggestion was meant to address?

- **About surface refinement and segmentation:** You mentioned that surfaces are refined, leading to an updated segmentation in aparc.DKTatlas+aseg.mapped.mgz. Would it be possible to prevent surface movement entirely (perhaps by skipping mri_make_surfaces in the recon process)? My hypothesis is that this would keep the final segmentation identical to the deep learning output, allowing WM parcellation to strictly follow the original segmentation without leaving any WM voxels unassigned. Please let me know if I am being wrong.

I understand this approach would sacrifice fine-tuned grey-white boundaries, but maintaining the original segmentation boundaries is more critical for my use case.


Your team is awesome btw

**Comment by m-reuter** [comment]
Yes, in FastSurfer `aparc+aseg.mgz` is just a sym-link to `aparc.DKTatlas+aseg.mapped.mgz`and therefore identical (as long as you don't use the --fsaparc flag, which will run FreeSurfers spherical atlas segmentation and produce the results of DK atlas). So it was my recommendation to use that file rather than the *.deep.mgz version. 

Keeping surfaces fixed to the original boundaries is not straight forward, they are for example smoothed and topology fixed which can change things already, even without the fine-tune placement. 

It is not clear to me why it is more critical for you to use the deep version instead without the surface based fine tuning?

**Comment by dat-rohit** [comment]
Thank you @m-reuter  for the clarification—it’s been incredibly helpful! To provide more context, I’m currently working on a kernel-based analytical method to refine segmentation ROIs immediately after the deep learning outputs. The goal of this research is to assess how these refined segmentations impact downstream processes, such as white matter lobe parcellation.

For this method to be properly evaluated, it is critical to maintain the boundaries and voxel-level segmentation output from the deep learning stage (aparc.DKTatlas+aseg.deep.mgz) without further refinement. Specifically, the surface-based fine-tuning introduces adjustments that could interfere with isolating the impact of my method on the white matter parcellation process.

I understand keeping surfaces fixed is not straight forward but I was hoping to get some hints on where I could begin if I wanted to achieve this. If things like smoothing are involved, it might indeed be unfeasible.

**Comment by m-reuter** [comment]
With "refine segmentation" you mean fine-tune them (e.g. to be even more accurate?) then what is the ground truth for that? And why not make the original network better (if it needs fixing, which I doubt) rather than such a post-processing step? Or do you mean further sub-segment them (I think not as you merge everything into lobes later anyway). So the motivation of what you are trying to do is still unclear. 

For the projection of cortex labels into the WM, FreeSurfer uses the surfaces. Otherwise things would be easy, e.g. you could do a proximity based filling, but it is not clear how well that would work. Might be OK, since you use larger lobular regions. You would first merge the cortex labels into the lobe labels and then fill the WM based on proximity. 

BUT, the correct way of obtaining good WM lobe parcels is to simply train FastSurfer with the additional WM labels. We have trained directly to segment FreeSurfers's WM regions (wmparc) on diffusion images and it worked pretty well. If there is demand for that, we might include it into our next training round where we plan to add additional labels anyway. 

The challenge with the surfaces is not really the smoothing (which could also be switched off) but the topology fixing. It is essential for the projection that surface normals are correct and point into the right direction. That step can probably not be omitted. Of course you could try but the question with this approach is, how much will the possibly incorrect surface normal affect your WM mappings (and that would also interfere with isolating the impact).

**Comment by dat-rohit** [comment]
I now understand the obstacles to achieving this. Will be closing this topic as my questions were mostly answered.

Btw, would be highly interested in that WM region segmentation model on diffusion images, let me know if I can contribute to this.


Thanks a lot!
